### PR TITLE
Allow using .exe files for mbed OS tools, instead of Python

### PR DESCRIPTION
**Pull request by janjongboom**
This is just an idea on getting Mbed CLI to work on a non-Python computer... It requires compilation of the tools in mbed-os/tools, which is annoying as it makes Mbed OS a lot larger. Opening the PR to fuel discussion.

@theotherjimmy @arekzaluski

### CODE REVIEW

**Comment by bremoran** [comment]
Can this be made to work on Mac OS X and Linux as well?

**Comment by janjongboom** [comment]
@bremoran Yeah, at least on macOS. pyinstaller works fine there. For Ubuntu I'm running into issues with libusb errors.

**Comment by theotherjimmy** [comment]
I probably won't get time to review this until Thursday.

File: mbed/mbed.py [hunk 1]
```diff
@@ -1294,6 +1295,14 @@ def get_tools_dir(self):
 
         return self._find_file_paths(paths, 'make.py')
 
+    def get_tools_command(self, script):
+        if os.name != 'nt' \
```
Comment (screamerbg) [inline mbed/mbed.py]: Could you please reverse the logic here, so standalone is the special case and the default is the .py case?

**Comment by arekzaluski** [comment]
I'll also try to find some time this week to test it within pyInstaller in Windows. I'll also talk with Eclipse IDE team because I know that they are interested in using pyInstaller to install Mbed CLI.

File: mbed/mbed.py (hunk 1 above)
Comment (screamerbg) [reply mbed/mbed.py]: Bump @janjongboom

**Comment by janjongboom** [comment]
@screamerbg Can we let this rest for a bit, until I've talked with tools team? Maybe a different approach is better.

@arekzaluski Did you talk to Eclipse IDE team? If so, what did they say?

**Comment by arekzaluski** [comment]
@janjongboom Eclipse team took different approach. They are using mbed.py directly and are shipping it isolated using https://conda.io/miniconda.html. I'll be creating an installer for Mbed Studio and all required tools this sprint.

**Comment by janjongboom** [comment]
@arekzaluski Does that allow installation without an internet connection? If so, that'd be a good way for the CLI installer...

**Comment by arekzaluski** [comment]
Yes, it does. It supports --offline mode. It's quite likely that I'll use it also in Mbed CLI for Windows installer if everything goes smoothly for Mbed Studio.

**Comment by andrewleech** [comment]
Another option to throw into the mix is briefcase:
https://github.com/pybee/briefcase

PyInstaller often runs into support issues, with packages not working properly once built into an exe. It could really add some extra support headaches to the mix.

Briefcase provides a means of packaging your python app (mped-cli) along with normal python executable and all dependencies into a normal installer (on windows) or a proper app bundle (on mac). Linux still uses the system python but it packages up all the dependencies to make it easier to distribute.

**Comment by janjongboom** [comment]
@BlackstoneEngineering ^ Interesting to look at for the MacOS installer?

**Comment by theotherjimmy** [comment]
@janjongboom see https://github.com/armmbed/mbed-os-tools 😄

**Comment by mark-edgeworth** [comment]
We are likely to eliminate the whole tool-calling-tool strategy soon, so this is no longer relevant.

**Pull request closed without merging**